server/db/dbManager.ts

```diff
@@ -24,15 +24,23 @@ export class PostManager {
 		return this.model.findById(postId).exec();
 	};
 
-	async getAllPosts(parent?: mongoose.Types.ObjectId): Promise<models.IPost[]> {
+	async getAllPosts(parent?: mongoose.Types.ObjectId, depth: number = 0, parentObject = { children: [] }): Promise<models.IPost[]> {
```

Comment:
```suggestion
	async getAllPosts(parent?: mongoose.Types.ObjectId, depth = 0, parentObject = { children: [] }): Promise<models.IPost[]> {
```

When you have a primitive default value, the type is redundant because it is determined automatically.